doc/sphinx/src/building.rst

```diff
@@ -3,3 +3,216 @@
 Building Singularity-EOS
 =========================
 
+There are two main ways to build singularity-eos:
+1. Through the spack interface
+2. Through the cmake system
+
+Installing Via Spack
+--------------------
+
+Although the spackage has not yet made it to the main `Spack`_
+repositories, we provide a spackage for ``singularity-eos`` witin the
+the singularity-eos source repository. If you have spack installed,
+simply call
+
+.. _Spack: https://spack.io/
+
+.. code-block:: bash
+  git clone --recursive git@github.com:lanl/singularity-eos.git
+  spack repo add singularity-eos/spack-repo
+  spack install singularity-eos
+
+to install ``singularity-eos`` into your spack instance. The spackage
+supports a number of relevant variants:
```

Comment:
Is the spackage ready for general use:
I tried: 
`spack install singularity-eos +kokkos +tests build_extra=sesame`
and got:
`py-h5py requires hdf5 variant +mpi, but spec asked for ~mpi`

Reply:
Is mpi an external dependency or being scheduled to build by spack? i.e. is there an external, non-buildable mpi in a package.yaml file? Also, is this in the context of an environment? there may be something in `$HOME/.spack/` that is conflicting here.

Reply:
Removing ~/.spack and adding +mpi concretized (will not concretize w/o +mpi) and currently building.

Reply:
I will update the spackage to conflict if `+sesame~mpi`, which seems to be causing the issue (or whatever else variant requires `py-hdf5`

Reply:
don't know why hdf5 requires MPI. It shouldn't. But... 🤷 python is always finicky.

Reply:
@Yurlungur I think `py-hdf5` is needed otherwise something doesn't work when we were first adding the CI?

Reply:
Right `py-hdf5` is needed for the `stellarcollapse` tests. What confuses me is that it needs `mpi`. There's no reason to build parallel hdf5 for python.

Reply:
Ah I think I found the problem. `py-h5py` has `mpi` on by default in its variant. Do we need to pass `~mpi` down through it?

Reply:
Ok, spack build w/ ~mpi fails with:
  ```
   131                     from /tmp/annap/spack-stage/spack-stage-singularit
            y-eos-main-brgwdvmhfkzipj2bjry2g2e5ws6em2hj/spack-src/singularity-e
            os/eos/singularity_eos.cpp:19:
  >> 132    /tmp/annap/spack-stage/spack-stage-singularity-eos-main-brgwdvmhfkz
            ipj2bjry2g2e5ws6em2hj/spack-src/singularity-eos/closure/mixed_cell_
            models.hpp:272:2: error: #error "Eigen should not be used with Kokk
            os."
     133      272 | #error "Eigen should not be used with Kokkos."
     134          |  ^~~~~
 
```

and 

```
     141    /usr/bin/ld: /projects/artab/users/annap/spack-environments/DWP9X/1
            .8.0/spack/opt/spack/linux-rhel7-power9le/gcc-9.3.0/eospac-6.4.2bet
            a-jxle75damh3twfcaicx7xdboktbyugk2/lib/libeospac6.a(eos_Interpolati
            on.o): In function `eos_BiRationalInterpolate':
     142    eos_Interpolation.c:(.text+0x520c): toc optimization is not support
            ed for 0xe4e50002 instruction.
     143    make[2]: Leaving directory `/ram/tmp/annap/spack-stage/spack-stage-
            singularity-eos-main-brgwdvmhfkzipj2bjry2g2e5ws6em2hj/spack-build-b
            rgwdvm'
     144    [ 73%] Built target sesame2spiner
  >> 145    make[2]: *** [singularity-eos/eos/CMakeFiles/eos_objs.dir/singulari
            ty_eos.cpp.o] Error 1
     146    make[2]: Leaving directory `/ram/tmp/annap/spack-stage/spack-stage-
            singularity-eos-main-brgwdvmhfkzipj2bjry2g2e5ws6em2hj/spack-build-b
            rgwdvm'
  >> 147    make[1]: *** [singularity-eos/eos/CMakeFiles/eos_objs.dir/all] Erro
            r 2
     148    make[1]: *** Waiting for unfinished jobs....

```

Reply:
@Yurlungur What was the status of this error? Did we ever decide what to do about it?

Reply:
um... I don't understand why we're hitting that error. It *should* only happen if you build with Kokkos. That said that macro is wrong. I will try to push a correction to this branch.